exporter/metric/metric.go

```diff
@@ -604,3 +604,15 @@ func aggToTypedValue(kind apimetric.NumberKind, value apimetric.Number) (*monito
 	}
 	return nil, errUnexpectedNumberKind{kind: kind}
 }
+
+// https://github.com/googleapis/googleapis/blob/c4c562f89acce603fb189679836712d08c7f8584/google/api/metric.proto#L149
+//
+// > The label key name must follow:
+// >
+// > * Only upper and lower-case letters, digits and underscores (_) are
+// >   allowed.
+// > * Label name must start with a letter or digit.
+// > * The maximum length of a label name is 100 characters.
+func normalizeLabelKey(s string) string {
+	return strings.ReplaceAll(s, ".", "_")
```

Comment:
This looks good as the first step, though Ideally we would also need to make sure that the label key is valid after this conversion (with all the other limitations you listed above), and might be useful to return an error if it's still invalid after "normalization".

Reply:
Yes. We try to adopt this exporter to our production and we might send another PR(s) in the future. Thanks for the quick merge 👍

Reply:
This is still relevant today. Especially with third-party library instrumentation, it's not always trivial to guarantee all metric names follow Google's conventions. For now I'm replacing all characters outside `[A-Za-z0-9]` with `_` and truncating to 100 characters